# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RendererScreenshotGrabber.cpp
# Commit: 971767c6ed2b
# Full Hash: 971767c6ed2b2201e172105046ecc61942e03a7e
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2020-01-24 21:46:30
# Regressor Bug: 1602643
# File Overlap Count: 1
# Description:
#   Bug 1602643 - Disable WebRender compositor dinamically for async screenshot r=gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D59658
# ==============================================================================

diff -r 436df7dc06c6 -r 971767c6ed2b gfx/webrender_bindings/RendererScreenshotGrabber.cpp
--- a/gfx/webrender_bindings/RendererScreenshotGrabber.cpp	Fri Jan 24 01:04:09 2020 +0000
+++ b/gfx/webrender_bindings/RendererScreenshotGrabber.cpp	Thu Jan 23 23:52:05 2020 +0000
@@ -16,27 +16,33 @@
 }
 
 void RendererScreenshotGrabber::MaybeGrabScreenshot(
-    Renderer* aRenderer, const gfx::IntSize& aWindowSize) {
-  if (ProfilerScreenshots::IsEnabled()) {
+    RendererOGL* aRendererOGL, const gfx::IntSize& aWindowSize) {
+  bool isEnabled =
+      ProfilerScreenshots::IsEnabled() && aRendererOGL->EnsureAsyncScreenshot();
+
+  if (isEnabled) {
     if (!mProfilerScreenshots) {
       mProfilerScreenshots = new ProfilerScreenshots();
     }
 
-    GrabScreenshot(aRenderer, aWindowSize);
+    GrabScreenshot(aRendererOGL->GetRenderer(), aWindowSize);
   } else if (mProfilerScreenshots) {
-    Destroy(aRenderer);
+    Destroy(aRendererOGL->GetRenderer());
   }
 }
 
-void RendererScreenshotGrabber::MaybeProcessQueue(Renderer* aRenderer) {
-  if (ProfilerScreenshots::IsEnabled()) {
+void RendererScreenshotGrabber::MaybeProcessQueue(RendererOGL* aRendererOGL) {
+  bool isEnabled =
+      ProfilerScreenshots::IsEnabled() && aRendererOGL->EnsureAsyncScreenshot();
+
+  if (isEnabled) {
     if (!mProfilerScreenshots) {
       mProfilerScreenshots = new ProfilerScreenshots();
     }
 
-    ProcessQueue(aRenderer);
+    ProcessQueue(aRendererOGL->GetRenderer());
   } else if (mProfilerScreenshots) {
-    Destroy(aRenderer);
+    Destroy(aRendererOGL->GetRenderer());
   }
 }
 